ex1/functions.py

```diff
@@ -90,26 +243,142 @@ def f():
 # Define a function named args_to_dict that takes three arguments and returns a
 # dictionary with the position of the argument as the key (starting at 0) and
 # the argument as the value.
+def args_to_dict(a0,a1,a2):
+	"""
+	Parameters
+	----------
+	a0,a1,a2 : optional
+
+	Returns
+	-------
+	dictionary of optional
+				Key values are int (starting at 0 to 2)
+	See also
+	--------
+	args_to_dict_general: take any number of arguments
+    	"""
+	return {0: a0,1: a1, 2: a2}
 
 # BONUS: Write a function named args_to_dict_general that does the same for any
 # number of arguments
+def args_to_dict_general(*a):
+	"""
+	Parameters
+	----------
+	a* : pointer to optional
+				point at any number of arguments
+
+	Returns
+	-------
+	dictionary of optional
+				Key values are int (starting at 0)
+
+	See also
+	--------
+	args_to_dict: take 3 arguments
+    	"""
+
+	d = {} #dictionary
+	ic = 0 # index count
+	for i in a:
+		d.update({ic: i})
+		ic+=1
+	return d
 
 # Define a function named lists_to_dict that takes two lists of equal lenght
 # named keys and values and builds a dictionary out of them.
+def lists_to_dict(l1,l2):
+	"""
+	Parameters
+	----------
+	l1,l2 : list of optional
+				equal lenght
+
+	Returns
+	-------
+	dictionary of optional
+				Keys are from l1, Values are from l2
+
+    	"""
+	len_ = len(l1)
+	d = {} # new empty dictionary
+	for i in range(len_):
+		d.update({l1[i]: l2[i]})
+	return d
 
 # Define a function named search_list that takes two lists a and b. The
 # function searches for all elements of b in list a. The return value should be
 # a dictionary containing the index of the found element of b in list a and the
 # value of the found element. If nothing was found then return an empty
 # dictionary.
+def search_list(la,lb):
+	"""
+	Parameters
+	----------
+	la,lb : list of optional
 
+	Returns
+	-------
+	dictionary of optional
+			Keys are the indexes of found elements. The values are the found elements.
+
+    	"""
+	lenA = len(la)
+	lenB = len(lb)
+	d = {}
+	for i in range(lenB):
+		for k in range(lenA):
+			if(lb[i] == la[k]):
+				d.update({k: lb[i]})
+	return d
 # Define a function named dict_to_string that takes a dictionary and a
 # separator string. The function should only take elements out of the
 # dictionary whose value is a string and then return a single string containing
 # the strings stored in the dictionary seperated by the separator string.
 # Return an empty string if there are no strings in the dictionary.
+def dict_to_string(d, s):
+	"""
+	Parameters
+	----------
+	d : dictionary
+	s : string
+		separator
 
+	Returns
+	-------
+	string
+		containing the strings stored in the dictionary seperated by the separator
+    	"""
+	str_ = ''
+	for key in d:
+		if(type(d[key]) == type("a")):
```

Comment:
type comparisons can use `str` or `int` directly. No need to provide an example type.